Makefile

```diff
@@ -113,16 +113,27 @@ srpm: .srpm-done
 
 rpm: .rpm-done
 
-ubuntu-trusty:
-	cp packaging/ubuntu-trusty/ecs.conf ecs.conf
-	tar -czf ./amazon-ecs-init_${VERSION}.orig.tar.gz ecs-init ecs.conf scripts README.md
+ARCH:=$(shell uname -m)
+ifeq (${ARCH},x86_64)
+	AGENT_URL=https://s3.amazonaws.com/amazon-ecs-agent/ecs-agent-v${VERSION}.tar
+else ifeq (${ARCH},aarch64)
+	AGENT_URL=https://s3.amazonaws.com/amazon-ecs-agent/ecs-agent-arm64-v${VERSION}.tar
+endif
+
+download-agent:
+	curl -o /tmp/ecs-agent-v${VERSION}.tar ${AGENT_URL}
+
+deb: download-agent
+	./scripts/update-version.sh
+	cp packaging/generic-deb/ecs.service ecs.service
+	tar -czf ./amazon-ecs-init_${VERSION}.orig.tar.gz ecs-init ecs.service scripts README.md
 	mkdir -p BUILDROOT
-	cp -r packaging/ubuntu-trusty/debian BUILDROOT/debian
+	cp -r packaging/generic-deb/debian BUILDROOT/debian
 	cp -r ecs-init BUILDROOT
-	cp packaging/ubuntu-trusty/ecs.conf BUILDROOT
+	cp packaging/generic-deb/ecs.service BUILDROOT
 	cp -r scripts BUILDROOT
 	cp README.md BUILDROOT
-	cd BUILDROOT && debuild $(shell [ "$(DEB_SIGN)" -ne "0" ] || echo "-uc -us")
+	cd BUILDROOT && debuild $(shell [ "$(DEB_SIGN)" -ne "0" ] || echo "-uc -us") --lintian-opts --suppress-tags bad-distribution-in-changes-file
```

Comment:
It'd probably be cleaner to store the "-uc -us" options in a variable, conditional on DEB_SIGN being set, and reference that variable here instead of invoking the shell. Maybe store the lintian flags in a variable, too. Something like
```
debuild_flags:=--lintian-opts --suppress-tags bad-distribution-in-changes-file
ifeq (${DEB_SIGN},0)
debuild_flags+= -uc -us
endif
...
deb:
        ...
        cd BUILDROOT && debuild ${debuild_flags}
        ...

Reply:
I'm not aware of a use-case where we actually need DEB_SIGN, so I will just remove it for now and we can add later if it's needed for building the official deb packages that we distribute.

Reply:
Yeah, I think that's reasonable. The signatures are attached to the source artifacts, not the binaries .deb files, and you're not currently doing anything with those anyway.